server/channels/app/platform/status.go

```diff
@@ -40,13 +41,36 @@ func (ps *PlatformService) GetAllStatuses() map[string]*model.Status {
 	}
 
 	statusMap := map[string]*model.Status{}
-	if userIDs, err := ps.statusCache.Keys(); err == nil {
-		for _, userID := range userIDs {
-			status := ps.GetStatusFromCache(userID)
-			if status != nil {
-				statusMap[userID] = status
+	err := ps.statusCache.Scan(func(keys []string) error {
+		if len(keys) == 0 {
+			return nil
+		}
+
+		toPass := make([]any, 0, len(keys))
+		for i := 0; i < len(keys); i++ {
+			var status *model.Status
+			toPass = append(toPass, &status)
+		}
+		errs := ps.statusCache.GetMulti(keys, toPass)
+		for i, err := range errs {
+			if err != nil {
+				if err != cache.ErrKeyNotFound {
+					return err
+				}
+				continue
```

Comment:
Are we tracking hit/miss metrics for this?

Reply:
Nope, we never did. This method isn't even guarateed to return all statuses since we are not falling back to DB for cache misses. This is only called via the wsapi, which I think is just a best effort thing. That's why I preserved the functionality.